### PR TITLE
Added the App Deploy card  and history list in the application tab .

### PR DESCRIPTION
**Pull request by hrushikesh07**
@revinjacob 
Added the App Deploy card view section and history list section in the application tab .

### CODE REVIEW

File: client/cat3/src/factory/workzoneServices.js [hunk 1]
```diff
@@ -62,9 +62,10 @@
                     var url = '/app/deploy/pipeline' + '/project/' + projId;
                     return $http.get(fullUrl(url), Auth.getHeaderObject());
                 },
-                getApplicationHistoryForEnv: function (envName, projId) {
-                    //var url = '/cat3/data/appDeployEnvList.json';
-                    var url = '/app/deploy/env/' + envName + '/project/' + projId + '/list';
+                getApplicationHistoryForEnv: function (envName, projId,pagiOpti) {
+                    var pageiReq='pageNumber='+pagiOpti.page+',pageSize='+pagiOpti.pageSize+',field='+pagiOpti.sortBy+',direction='+pagiOpti.sortOrder;
```
Comment (revinjacob) [inline client/cat3/src/factory/workzoneServices.js]: If a smaller variable name is the intention, please rename this to pgOptions

File: client/cat3/src/factory/workzoneServices.js [hunk 2]
```diff
@@ -402,6 +403,15 @@
                     var p = workzoneEnvironment.getEnvParams(),
                         url = '/app/deploy/nexus/'+requestData.nexus+'/repositories/'+requestData.repositories+'/group/'+requestData.group+'/artifact/'+requestData.artifactId+'/versionList';
                     return $http.get(fullUrl(url));
+                },
+                getPipelineConfig:function(requestEnv){
+                   var url = '/app/deploy/pipeline/project/'+requestEnv.proj;
+                    return $http.get(fullUrl(url));
+                },
+                getPipelineView :function(requestEnv,pagiOpti){
+                    var pageiReq='page='+pagiOpti.page+'&pageSize='+pagiOpti.pageSize+'&sortBy='+pagiOpti.sortBy+'&sortOrder='+pagiOpti.sortOrder;
```
Comment (revinjacob) [inline client/cat3/src/factory/workzoneServices.js]: There is a pagination utility function which does the url creation. So this logic can be removed from here

File: client/cat3/src/partials/sections/dashboard/workzone/application/application.html [hunk 3]
```diff
@@ -34,105 +33,29 @@
...
-                            </div>
-                            <div class="card-status-state">
-                                <div class="app-instance-image"></div>
-                            </div>
-                            <div class="instance-ip">
-                                <span>{{row.applicationVersion}}</span>
-                            </div>
-                        </div>
+        <section id="pipelineView" class="scrollContent" ng-show="isPipelineViewActive" >
+            <div ng-if="PipeGridOptions.data.length > 0">
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/application/application.html]: Variable names should follow lower camel case format, ie start with lower case letter.

File: client/cat3/src/partials/sections/dashboard/workzone/application/application.html [hunk 4]
```diff
@@ -34,105 +33,29 @@
...
-                                    {{part}}
-                                    <br>
-                                </span>
-                            </div>
-                        </div>
-                    </div>
-                </li>
-            </ul>
+                </div>
+                <div id="grid1"  ui-grid="PipeGridOptions"  ui-grid-pagination class="grid"></div>
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/application/application.html]: Is id attribute required? If required, change it to convey context and meaning

File: client/cat3/src/partials/sections/dashboard/workzone/application/application.html [hunk 5]
```diff
@@ -34,105 +33,29 @@
...
-                        <td>
-                            <i class="fa fa-info-circle cursor" title="More Info" ng-click="appInfo(app)"></i>
-                        </td>
-                        <td>
-                            <i class="fa fa-cloud-upload cursor" title="Upgrade App"></i>
-                        </td>
-                    </tr>
-                </tbody>
-            </table>
+        <section id="tableView" class="scrollContent" style="position: relative;" ng-show="!isPipelineViewActive">
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/application/application.html]: remove inline style

File: client/cat3/src/partials/sections/dashboard/workzone/application/application.html [hunk 6]
```diff
@@ -34,105 +33,29 @@
...
+            <div class="no-data-found" ng-if="!historyGridOptions.data.length > 0"> No data found !</div>
+            <div id="grid2" ng-if="historyGridOptions.data.length > 0"  ui-grid="historyGridOptions" ui-grid-pagination  ui-grid-resize-columns ui-grid-move-columns  class="grid"></div>
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/application/application.html]: Is id attribute required? If required, change it to convey context and meaning

File: client/cat3/src/partials/sections/dashboard/workzone/application/application.scss [hunk 7]
```diff
@@ -108,4 +108,39 @@
     .margintop20 {
         margin-top: 20px;
     }
+}
+
+
+#pipelineView{
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/application/application.scss]: This is good separation of pipeline view css.

File: client/cat3/src/partials/sections/dashboard/workzone/application/applicationCtrl.js [hunk 8]
```diff
@@ -191,36 +103,175 @@
...
+			function getApplicationPipeLineData(envParams) {
+				workzoneServices.getPipelineConfig(envParams).then(function(configResult){
+					$scope.PipeGridOptions=angular.extend(appData.gridOption,{enableSorting: false},{
+						onRegisterApi: function(gridApi) {
+							gridApi.pagination.on.paginationChanged($scope, function (newPage, pageSize) {
+								$scope.pagiOptionsCard.page=newPage;
+								$scope.pagiOptionsCard.pageSize=pageSize;
+								getApplicationCardService(envParams,$scope.pagiOptionsCard);
+								$scope.pipeLineActBarShow=false;
+								angular.element('.card').removeClass('selected-card');
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/application/applicationCtrl.js]: This css reference would affect globally. Make it modularized to affect only pipeline view. 
As an alternative to directly manipulating styles, use a selected-card modal object which applies the right classes.

**Comment by revinjacob** [comment]
@hrushikesh07 , 

Looks neat overall. Logic is segregated to proper methods. 
Introduction of directives for app-card is a good approach which can be followed in other places too.

Please take care of review comments.

File: client/cat3/src/factory/workzoneServices.js (hunk 1 above)
Comment (hrushikesh07) [inline client/cat3/src/factory/workzoneServices.js]: After merging the  utility file i will do this things .

**Pull request merged**